jumpscale/sals/marketplace/apps_chatflow.py

```diff
@@ -221,27 +221,37 @@ def _get_domain(self):
                     "Letsencrypt limit has been reached on all gateways. The resources you paid for will be re-used in your upcoming deployments."
                 )
 
-        self.domain = random.choice(list(domains.keys()))
+        self.addresses = []
 
-        self.gateway = domains[self.domain]["gateway"]
-        self.gateway_pool = domains[self.domain]["pool"]
+        while not self.addresses and domains:
+            managed_domain = random.choice(list(domains.keys()))
+            self.gateway = domains[managed_domain]["gateway"]
+            self.gateway_pool = domains[managed_domain]["pool"]
 
-        solution_name = self.solution_name.replace(f"{self.solution_metadata['owner']}-", "").replace("_", "-")
-        owner_prefix = self.solution_metadata["owner"].replace(".3bot", "").replace(".", "").replace("_", "-")
-        solution_type = self.SOLUTION_TYPE.replace(".", "").replace("_", "-")
-        # check if domain name is free or append random number
-        full_domain = f"{owner_prefix}-{solution_type}-{solution_name}.{self.domain}"
-        while True:
-            if j.tools.dnstool.is_free(full_domain):
-                self.domain = full_domain
-                break
-            else:
-                random_number = random.randint(1000, 100000)
-                full_domain = f"{owner_prefix}-{solution_type}-{solution_name}-{random_number}.{self.domain}"
+            solution_name = self.solution_name.replace(f"{self.solution_metadata['owner']}-", "").replace("_", "-")
+            owner_prefix = self.solution_metadata["owner"].replace(".3bot", "").replace(".", "").replace("_", "-")
+            solution_type = self.SOLUTION_TYPE.replace(".", "").replace("_", "-")
+            # check if domain name is free or append random number
+            full_domain = f"{owner_prefix}-{solution_type}-{solution_name}.{managed_domain}"
+            while True:
+                if j.tools.dnstool.is_free(full_domain):
+                    self.domain = full_domain
+                    break
+                else:
+                    random_number = random.randint(1000, 100000)
+                    full_domain = f"{owner_prefix}-{solution_type}-{solution_name}-{random_number}.{self.domain}"
```

Comment:
need  to use `manged_domain` instead of `self.domain` here

Reply:
righ, thanks